### PR TITLE
Upgrade null safety to latest flutter stable

### PR DESCRIPTION
**Pull request by remonh87**
@werediver  let's ship the first version. After this I want to extend the example app with read and write and subscribe (only for null safe) which would help us in triaging bugs and improving the API's from there. 

- Part of this pr is rebase from master
-  The following commits are new: [link1](https://github.com/PhilipsHue/flutter_reactive_ble/pull/225/commits/8c5ae7149d6d2035db660a88b7cbb8b8a6cffee5) [link2](https://github.com/PhilipsHue/flutter_reactive_ble/pull/225/commits/90d463f78d7fe718bc26c58294a14fc26a24dea1)

### CODE REVIEW

File: CHANGELOG.md
Comment (werediver) [inline CHANGELOG.md:1]: Why does this title appear twice in the file? I guess you wanted to move it, not copy.

Separately from that, why the null-safe version is set to 1.0.0? It looks like it's forked from a very old version of the library. I'd rather set it to 3.0.0-nullsafety.

**Review by werediver — COMMENTED** [review]
- Please check my comment,
- Clarify what do you mean by "let's ship"
- Open a ticket for extending the example app (or let's discuss it via an other channel) with read/write/subscribe to discuss some of the details (we may try to target some standard or widespread BLE service, like time service [which may be present on a phone by default])

File: CHANGELOG.md
Comment (remonh87) [reply CHANGELOG.md:1]: ah yess very good one I will change this

**Comment by remonh87** [comment]
@werediver  fixed the comments and also creted #226 . With ship it I mean: After merging this I will publish first prerelease version to pub.dev so users can try it out. Would be great to have some feedback from community on this impactful change.

**Review by werediver — APPROVED** [review]
Who doesn't like null-safety? 😄

**Pull request merged**
